### PR TITLE
Include full lineage in breadcrumbs

### PR DESCRIPTION
**Pull request by esoergel**
https://github.com/dimagi/commcare-hq/pull/6339#discussion_r28553545
@czue @sravfeyn
Also add it to new locations, which gives nice context to the new location screen (THIS is where the location is).  Even if it runs past the view width, the breadcrumbs wrap and it looks good (although you'd need some seriously nested locations and/or long names for that).

### CODE REVIEW

File: corehq/util/__init__.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import re
 
+from .couch import get_document_or_404
```
Comment (dimagimon) [inline corehq/util/__init__.py]: F401 'get_document_or_404' imported but unused

File: corehq/apps/locations/views.py [hunk 2]
```diff
@@ -250,19 +250,34 @@ class NewLocationView(BaseLocationView):
 
     @property
     def parent_pages(self):
-        return [{
+        selected = self.location._id or self.location.parent_id
```
Comment (esoergel) [inline corehq/apps/locations/views.py]: If you click "Locations" from the new location screen, jump to the parent's position in the tree.

File: corehq/util/__init__.py (hunk 1 above)
Comment (czue) [inline corehq/util/__init__.py]: fwiw, it's not clear to me that we're consistent enough with defining our module apis that these types of changes are actually net positive, but they don't bother me enough to fight either.

**Comment by czue** [comment]
:+1: from me (though didn't review too closely)

File: corehq/util/__init__.py (hunk 1 above)
Comment (esoergel) [inline corehq/util/__init__.py]: True.  I think that's a good direction to head in though.  If we decide to stick with full-path imports, I'd be fine with following that convention.

File: corehq/apps/locations/views.py [hunk 3]
```diff
@@ -250,19 +250,34 @@ class NewLocationView(BaseLocationView):
...
+        return breadcrumbs
 
     @property
     @memoized
     def location(self):
-        return Location(domain=self.domain, parent=self.parent_id)
+        parent_id = self.request.GET.get('parent')
+        parent = (get_document_or_404(Location, self.domain, parent_id)
+                  if parent_id else None)
+        return Location(domain=self.domain, parent=parent)
```
Comment (sravfeyn) [inline corehq/apps/locations/views.py]: Does the Location model handle if parent is `Http404()`?

**Pull request merged**